### PR TITLE
[BEAM-6099] RedisIO support for PFADD operation

### PR DESCRIPTION
**Pull request by varundhussa**
Add PFADD method to RedisIO. Add test for verifying PFADD.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java [hunk 1]
```diff
@@ -186,6 +186,25 @@ public void testWriteReadUsingRpushMethod() throws Exception {
...
   }
 
+  @Test
+  public void testWriteUsingHLLMethod() throws Exception {
+    String key = "key";
+    String values[] = {"1", "2"};
+
+    Jedis jedis =
+        RedisConnectionConfiguration.create(REDIS_HOST, embeddedRedis.getPort()).connect();
+    jedis.pfadd(key, values[0]);
```
[the remark this answers is not in the document]
Comment (varundhussa) [reply sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java]: It isn't required. I'll move it into the pipeline

File: sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java [hunk 2]
```diff
@@ -186,6 +186,25 @@ public void testWriteReadUsingRpushMethod() throws Exception {
...
+
+    PCollection<KV<String, String>> write = writePipeline.apply(Create.of(KV.of(key, values[1])));
```
[the remark this answers is not in the document]
Comment (varundhussa) [reply sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java]: That sounds good. Do you think adding random strings would make sense (would require a tolerance error range in count), or a known list (eg: integer to string). Thanks

[the remark this answers is not in the document]
Comment (aromanenko-dev) [reply sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java]: I'd suggest just to replicate the same test case as official Redis documentation provides for [pfcount](https://redis.io/commands/pfcount) command.

**Comment by varundhussa** [comment]
Thanks @aromanenko-dev. I'll raise a new PR with the required changes.

**Pull request closed without merging**